docs/exchange-operators/rosetta/build-from-sources.mdx

```diff
@@ -0,0 +1,60 @@
+---
+title: Building from sources
+description: Build Mina’s implementation of Rosetta API from sources.
+keywords:
+  - historical data
+  - blockchain integration
+  - explorer
+  - wallet
+  - archive node
+  - graphQL
+  - rosetta
+  - nix
+---
+
+
+# Building and running Rosetta from source code
+
+In the [previous article](run-with-docker) we described the Docker cointainer, that runs mina daemon, mina-archive and Rosetta API. 
+
+If you already have mina-archive up and running, you can also build and run Rosetta API natively and wire it to existing mina daemon and archive.
+
+:::note
+
+To run mina daemon and archive natively, follow instructions on the [Archive Node page](node-operators/archive-node)
+
+:::
+
+The easiest way to build Rosetta natively is to use Nix development environment.
+
+1. Clone the official [mina repository](https://github.com/MinaProtocol/mina.git) and switch to the `compatible` branch
+
+2. If you don't already have Nix on your machine, install it following steps described in the `nix/README.md` [file in the mina repository](https://github.com/MinaProtocol/mina/blob/develop/nix/README.md)
+
+3. Run the `./nix/pin.sh` script to enable submodules to be available to the build:
+
+
+    ./nix/pin.sh
+
+
+4. Launch the development shell:
+
+
+    nix develop mina
```

Comment:
this Markdown syntax seems to work, if you want to follow the established style, apply the syntax described in https://github.com/o1-labs/docs2/wiki/Markdown-Syntax#numbered-steps

Reply:
@barriebyron it doesn't work when numbered steps are interrupted with code snippets

but I willl change it in another places. Thank you

Reply:
fwiw @qwadratic auto-numbering does work, even with code snippets 
https://github.com/o1-labs/docs2/wiki/Markdown-Syntax#numbered-steps 

TLDR;  indent snippets

> To generate an ordered list, start each step with 1. so that each step automatically increments numbering. Auto-numbering is respected only when the content under a step is indented (2 or 4 spaces; in IDE such as VS Code, press the tab key).
> 
> The indentation for numbered steps is required to position the content under each step and to keep the auto-numbering in step.